### PR TITLE
[codex] Fail incomplete CLI output-dir workflows

### LINKED ISSUE #4269: CLI workflow --output-dir can succeed after copying only one multi-input output
## Context

A fresh built CLI probe found that `texra run correct` can accept multiple input files with `--output-dir`, exit successfully, and copy only one of the expected outputs.

## Reproduction

From the repository root after `corepack pnpm --filter @texra/cli build`:

```sh
TMPDIR=$(mktemp -d /tmp/texra-output-dir-workflow-XXXXXX)
mkdir -p "$TMPDIR/out"
printf '%s\n' '\documentclass{article}' '\begin{document}' 'The first sentence has a typoo.' '\end{document}' > "$TMPDIR/a.tex"
printf '%s\n' '\documentclass{article}' '\begin{document}' 'The second sentence has a mispelled word.' '\end{document}' > "$TMPDIR/b.tex"
node packages/cli/dist/bin/texra.js --cwd "$TMPDIR" run correct \
  --input a.tex \
  --input b.tex \
  --output-dir out \
  --model deepseekT \
  --instruction "Only fix typoo to typo and mispelled to misspelled. Preserve all other text." \
  --output-format text
find "$TMPDIR/out" -type f -maxdepth 3 -print
```

## Actual Result

The command exited with status `0`, stdout contained only:

```text
/private/tmp/texra-output-dir-workflow-Ey1iqM/out/a.tex
```

The output directory contained only `out/a.tex`. `out/b.tex` was not written.

## Expected Result

For a successful multi-input workflow run with `--output-dir`, every input file should have a corresponding copied output, or the command should fail with a clear error naming the missing output files. A successful exit with partial copied output is misleading and makes scripts treat an incomplete workflow as complete.

## Notes

The same probe showed progress on stderr and the final copied path on stdout, so the defect is specifically in output completeness validation/copying, not in stdout/stderr separation.

### PR DESCRIPTION
**Pull request by LionSR**
## Summary

- Make `texra run --output-dir` validate that the expected workflow outputs were actually copied.
- Derive expected output names from the agent's default outputs, or from the input filenames for rewrite-style workflow agents such as `correct` and `polish`.
- Deduplicate repeated round outputs by copied relative path and keep the latest round's file.
- Add focused tests for missing expected outputs, duplicate round outputs, and the existing interrupted-output path.

## Root Cause

The CLI copied whatever workflow outputs the runtime returned, then exited successfully as long as at least one output existed. For multi-input rewrite workflows this allowed a partial result, for example only `a.tex` for inputs `a.tex` and `b.tex`, to be treated as a successful `--output-dir` run.

## Validation

- `corepack pnpm exec vitest run --config vitest.config.mjs src/test-kernel/cli/WorkflowOutputResolution.vitest.mts src/test-kernel/cli/CliRootArgs.vitest.mts src/test-kernel/cli/TuiStateAndFocus.vitest.mts`
- `corepack pnpm --filter @texra/cli typecheck`
- `corepack pnpm --filter @texra/cli build`
- `npm run typecheck`
- `npm run lint` (18 existing warnings, 0 errors)
- `npm run compile:fast`
- `npm run format`
- Built CLI probe with the internal validation model: two input files, one produced document, `--output-dir`; command exited `1` and reported `Workflow completed without expected outputs: a.tex, b.tex; copied 1 of 2 expected outputs ...`

Closes #4269.

---

> [!NOTE]
> **Medium Risk**
> Changes CLI workflow output copy/validation behavior and exit codes for `--output-dir`, which can cause previously successful runs to fail when outputs are incomplete. Logic is contained to CLI output resolution and covered by new focused tests, keeping overall risk moderate.
> 
> **Overview**
> **`texra run --output-dir` now validates completeness of copied workflow outputs.** The CLI derives an expected output set (from an agent’s `defaultOutputFiles` or falling back to input filenames) and throws an error if any expected outputs are missing, instead of succeeding after copying a partial set.
> 
> When copying to `--output-dir`, outputs are **deduplicated by normalized relative path** and the **latest `round` wins** to avoid stale overwrites. Error handling was tweaked so interrupted runs preserve `Interrupted` exit behavior, and new vitest coverage was added for missing outputs and multi-round dedupe.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 965712db9dc23bc9c11aeb290cacd1c367bb57fd. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by LionSR** [comment]
should we retire default outputs?

**Comment by LionSR** [comment]
Not to over design the systems

File: packages/cli/src/commands/root.ts [hunk 1]
```diff
@@ -995,21 +1019,36 @@ export async function resolveWorkflowOutput(
...
   if (outputDir && result.category === AgentCategory.Workflow) {
     const targetRoot = path.isAbsolute(outputDir)
       ? outputDir
       : path.join(context.cwd, outputDir);
-    const copiedOutputs: string[] = [];
+    const outputsByRelativePath = new Map<string, OutputFileSummary>();
     for (const output of result.outputs) {
-      const targetPath = path.join(targetRoot, outputCopyRelativePath(output));
+      outputsByRelativePath.set(outputCopyRelativePath(output), output);
+    }
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/commands/root.ts:1044]: Addressed in 965712db9: `--output-dir` de-duplication now keeps the output with the greatest workflow `round` for each copied relative path, independent of result iteration order. Added regression coverage for out-of-order older/newer outputs.

File: packages/cli/src/commands/root.ts [hunk 2]
```diff
@@ -995,21 +1019,36 @@ export async function resolveWorkflowOutput(
...
     }
 
+    const expectedOutputFiles = options.expectedOutputFiles ?? [];
+    const missing = expectedOutputFiles
+      .map(expectedOutputCopyRelativePath)
+      .filter((expected) => !outputsByRelativePath.has(expected));
+    if (missing.length > 0) {
+      throw new Error(
+        `Workflow ${terminalStatus} without expected output${missing.length === 1 ? '' : 's'}: ${missing.join(', ')}; copied ${copiedOutputs.length} of ${expectedOutputFiles.length} expected output${expectedOutputFiles.length === 1 ? '' : 's'} to ${targetRoot}.`,
+      );
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/commands/root.ts:1061]: Addressed in 965712db9: output resolution failures after an interrupted workflow now return `CliExitCode.Interrupted` and no longer rewrite the terminal status to error.
